spec.html

```diff
@@ -430,19 +388,41 @@ <h1>InnerModuleEvaluation( _module_, _stack_, _index_ )</h1>
       </emu-alg>
       <emu-note>
         <p>A module is `"evaluating"` while it is being traversed by InnerModuleEvaluation. A module is `"evaluated"` when ModuleExecution has been called, even if that module execution is a promise that has not yet resolved.</p>
-      </emu-nnote>
+      </emu-note>
     </emu-clause>
 
+  <emu-clause id="sec-execute-module-when-imports-ready" aoid="ExecuteModuleWhenImportsReady">
+    <h1><ins>ExecuteModuleWhenImportsReady( _module_, _promises_, _capability_ )</ins></h1>
+    <emu-alg>
+      1. If _promises_ is an empty List,
+        1. Perform ! ModuleExecution(_module_, _capability_).
+        1. Return.
+      1. Let _index_ be 0.
+      1. Let _fullfilledCount_ be 0.
```

Comment:
Could abstract this machinery as a spec-based `Promise.all` since it is so common?

Reply:
Maybe? I would rather wait for a second use case to come along so I can figure out what is needed in general.

Reply:
Sure, that is understandable. If I recall this pattern already exists in some other async spec code, but I can't recall where exactly.

Reply:
Within the JS spec, or in higher-level specifications? I have only seen it in the W3C Promise guide.

I doubt the W3C Promise guide would reference a Stage 2 draft; maybe we could make this editorial change at some later point (e.g., when integrating into the main spec) when it would make sense for that reference to be added.

Reply:
Actually I think the only other similar pattern is in the implementation of Promise.all itself :)